cocos2d/audio/CCAudio.js

```diff
@@ -123,6 +123,8 @@ Audio.State = {
     proto._bindEnded = function (callback) {
         callback = callback || this._onended;
         if (this._audioType === Audio.Type.DOM) {
+            this._element.removeEventListener('ended', this.__endedCallback);
```

Comment:
if (this.__endedCallback) {
this._element.removeEventListener('ended', this.__endedCallback);
}